python/config_settings/BUILD.bazel

```diff
@@ -10,7 +10,29 @@ filegroup(
     visibility = ["//python:__pkg__"],
 )
 
-construct_config_settings(
-    name = "construct_config_settings",
-    python_versions = TOOL_VERSIONS.keys(),
+string_flag(
+    name = "python_version",
+    # TODO: The default here should somehow match the MODULE config. Until
+    # then, use the empty string to indicate an unknown version. This
+    # also prevents version-unaware targets from inadvertently matching
+    # a select condition when they shouldn't.
+    build_setting_default = "",
+    values = [""] + VERSION_FLAG_VALUES.keys(),
+    visibility = ["//visibility:public"],
 )
+
+[
+    is_python_config_setting(
+        name = "is_python_{}".format(version),
+        flag_values = {":python_version": version},
+        match_extra = [
+            # Use the internal labels created by this macro in order to handle matching
+            # 3.8 config value if using the 3.8 version from MINOR_MAPPING. If we used the
+            # public labels we would have a circular dependency loop.
+            ("_is_python_{}" if VERSION_FLAG_VALUES[x] else "is_python_{}").format(x)
```

Comment:
note: This is slightly smelly, but I am not sure if we can do anything better here. I do like having the definition of the `python_version` flag here better and using a more explicit way to define the public config setting values is also an improvement over the previous structure.

If we don't want to have this comment here, we can keep this in an internal function.